Libraries/Network/NetInfo.js

```diff
@@ -233,14 +230,25 @@ const NetInfo = {
     },
   },
 
-  isConnectionExpensive(callback: (metered: ?boolean, error?: string) => void): void {
+  isConnectionExpensive(): Promise {
+    if (arguments.length > 1) {
+      console.warn('NetInfo.isConnectionMetered(callback) is deprecated. Use the returned Promise instead.');
+      const callback = arguments[1];
+      if (Platform.OS === 'android') {
+        RCTNetInfo.isConnectionMetered().then(
+          metered => callback(metered),
+          error => callback(null, error)
+        );
+      } else {
+        callback(null, 'Unsupported');
+      }
+      return;
```

Comment:
I left it as it is because this code for handling callbacks is temporary, and will be removed in next release. Promisify would be great if we want to support both promises and callbacks, but as far as I understand, we're only going to support promises.

Good thing about the completely separated if else is, we can just remove that block in the future and everything will work fine. If I write some logic to handle both promises and callbacks, then I have to undo all those changes when we entirely remove support for callbacks.